torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1624,11 +1623,11 @@ Tensor renorm_jvp(
 
   return where(
       norm > double_maxnorm,
-      factor *
-          (self_t -
-           self_p * invnorm *
-               norm_jvp(
-                   self_p, self_t, p, norm, reduce_dims, /*keepdim=*/true)),
+      (factor *
+       (self_t -
+        self_p * invnorm *
+            norm_jvp(self_p, self_t, p, norm, reduce_dims, /*keepdim=*/true)))
+          .to(self_t.scalar_type()),
```

Comment:
Why this change is needed now?

Reply:
The behaviors for renorm forward and backward are a bit different:

In forward, the `acc_type` is fixed to the ones for cuda:

https://github.com/pytorch/pytorch/blob/1aacbaed8b7b85afe93d039fd27e2b1d2f77bb8f/aten/src/ATen/native/Normalization.cpp#L858-L861

In backward, the `acc_type` depends on the device of the input tensor:

https://github.com/pytorch/pytorch/blob/1aacbaed8b7b85afe93d039fd27e2b1d2f77bb8f/torch/csrc/autograd/FunctionsManual.cpp#L1565-L1569

To be honest I'm not sure why the difference presents but I feel the backward one would be more appropriate. So I aligned the behavior of `acc_type` for `renorm_jvp` with the one in backward, and therefore if the `acc_type` is different from the original type, we should cast the type back, hence this change.

Reply:
That sounds like a dangerous change we would need to review super carefully. I think it would be best to land the cleanup and do this change in a follow up. WDYT?

Reply:
Sure.